### ISSUE TITLE
Fix loading models too eagerly (i.e. before registering with funcx)

### ISSUE THREAD

**Issue by OwenPriceSkelly** [Garden-AI/garden#93]
_The current implementation of `garden_ai.Model(...)` eagerly downloads the MLFlow model with a given URI, and python's "default arguments are evaluated once when defined" behavior gets us some caching behavior for free. The pre-downloaded object is also currently inspected by the `Step` decorating it, in order to extract model-specific dependencies_

However, because it's the default value for one of the function's arguments, the model is now considered part of the function's definition; when the function definition is serialized and registered with/ uploaded to funcx, _the entire `Model` object is included_ as part of the definition. 

Note that this only applies to `garden_ai.Model(...)` expressions that appear as default args -- a model that is only downloaded in the body of a function is not a problem here. 

tldr: steps can infer model-level dependencies at definition time, because the model is part of the definition (yay), but steps also load the entire model into memory at definition time, because it's part of the definition (boo)

# Assumptions:
1. the call to `garden_ai.Model(...)` should not download the entire model unnecessarily 
2. the step still needs to infer the model-specific dependencies correctly 
3. when registering a pipeline, the model should not be uploaded 

# Acceptance Criteria
Given..., when..., then...
